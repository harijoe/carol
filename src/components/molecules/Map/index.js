import React, { PropTypes, Component } from 'react'
import GoogleMap from 'google-map-react'

import { google } from 'config'
import { Marker } from 'components'

const styles = {
  position: 'relative',
  width: '100%',
  height: '300px',
}

class Map extends Component {
  static propTypes = {
    initialCenter: PropTypes.object,
    zoom: PropTypes.number,
    markers: PropTypes.array,
    onMarkerClick: PropTypes.func,
    country: PropTypes.string.isRequired,
  }

  static defaultProps = {
    initialCenter: {
      FR: {
        lat: 46.938698,
        lng: 2.399887,
      },
      GB: {
        lat: 54.355481,
        lng: -2.278986,
      },
      ES: {
        lat: 40.481755,
        lng: -3.761941,
      },
    },
    zoom: 5,
  }

  render() {
    const { initialCenter, zoom, markers, onMarkerClick, country } = this.props

    return (
      <GoogleMap
        defaultCenter={initialCenter[country]}
        defaultZoom={zoom}
        style={styles}
        bootstrapURLKeys={{ key: google.mapsKey }}
        onChildClick={onMarkerClick}
      >
        {markers.map((marker, i) => (
          <Marker
            key={i}
            lat={marker.position.lat}
            lng={marker.position.lng}
            title={marker.title}
            trade={marker.trade}
          />
        ))}
      </GoogleMap>
    )
  }
}

export default Map
