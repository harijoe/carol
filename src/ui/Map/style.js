const style = {
  map: {
    position: 'relative',
    width: '500px',
    height: '300px'
  }
}

export default style
