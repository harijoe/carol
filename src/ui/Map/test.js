import chai from 'chai'
import { shallow } from 'enzyme'
import React from 'react'
import Map from './'

const expect = chai.expect

describe('Map', () => {
  const enzymeWrapper  = shallow(<Map />)

  it('should have one google map', () => {
    expect(enzymeWrapper.find('div')).to.have.length(2)
  })
})
