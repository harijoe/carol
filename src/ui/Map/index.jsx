import React from 'react'
import { Map, Marker, InfoWindow, GoogleApiWrapper as googleApiWrapper } from 'google-maps-react'
import { getCurrentCountry } from '../../utils/locale'
import config from '../../config'
import style from './style'

class GoogleMap extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      initialCenter: {
        FR: {
          lat: 46.938698,
          lng: 2.399887
        },
        GB: {
          lat: 54.355481,
          lng: -2.278986
        },
        ES: {
          lat: 40.481755,
          lng: -3.761941
        }
      },
      selectedPlace: {
        name: 'default'
      }
    }
  }

  render() {
    return (
      <Map
        google={window.google}
        initialCenter={this.state.initialCenter[getCurrentCountry()]}
        zoom={5}
        containerStyle={style.map}
      >
        {this.props.markers.map((marker, i) => {
          return (
            <Marker
              key={i}
              id={i}
              name={marker.title}
              position={{ lat: marker.position.lat, lng: marker.position.lng }}
              onClick={this.props.onMarkerClick}
            />
          )
        })}

        <InfoWindow>
          <div>
            <h1>{this.state.selectedPlace.name}</h1>
          </div>
        </InfoWindow>
      </Map>
    )
  }
}

GoogleMap.propTypes = {
  markers: React.PropTypes.array,
  onMarkerClick: React.PropTypes.func
}

export default googleApiWrapper({ apiKey: config.googleMapsApiKey, version: 3.25 })(GoogleMap)
