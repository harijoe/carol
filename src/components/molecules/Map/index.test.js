import React from 'react'
import { shallow } from 'enzyme'
import Map from '.'

const wrap = (props = {}) => shallow(<Map {...props} country="FR"/>)

it('renders the Map component with markers', () => {
  const wrapper = wrap({ markers: [{ position: { lat: 48.857767, lng: 2.293604 } }] })

  expect(wrapper).toMatchSnapshot()
})
