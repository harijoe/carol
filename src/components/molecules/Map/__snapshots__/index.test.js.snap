exports[`test renders the Map component with markers 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <GoogleMap
    bootstrapURLKeys={
        Object {
            "key": "AIzaSyCYrFIvVAYqnWCNjjAfGIanmwxVvj1FPj8",
          }
    }
    debounced={true}
    defaultCenter={
        Object {
            "lat": 54.355481,
            "lng": -2.278986,
          }
    }
    defaultZoom={5}
    distanceToMouse={[Function]}
    googleMapLoader={[Function]}
    hoverDistance={30}
    layerTypes={Array []}
    onChildClick={undefined}
    options={[Function]}
    style={
        Object {
            "height": "300px",
            "position": "relative",
            "width": "100%",
          }
    }
    yesIWantToUseGoogleMapApiInternals={false}>
    <Mock
        lat={48.857767}
        lng={2.293604}
        title={undefined}
        trade={undefined} />
</GoogleMap>,
  "nodes": Array [
    <GoogleMap
      bootstrapURLKeys={
            Object {
                  "key": "AIzaSyCYrFIvVAYqnWCNjjAfGIanmwxVvj1FPj8",
                }
      }
      debounced={true}
      defaultCenter={
            Object {
                  "lat": 54.355481,
                  "lng": -2.278986,
                }
      }
      defaultZoom={5}
      distanceToMouse={[Function]}
      googleMapLoader={[Function]}
      hoverDistance={30}
      layerTypes={Array []}
      onChildClick={undefined}
      options={[Function]}
      style={
            Object {
                  "height": "300px",
                  "position": "relative",
                  "width": "100%",
                }
      }
      yesIWantToUseGoogleMapApiInternals={false}>
      <Mock
            lat={48.857767}
            lng={2.293604}
            title={undefined}
            trade={undefined} />
</GoogleMap>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Map
        initialCenter={
                Object {
                        "ES": Object {
                          "lat": 40.481755,
                          "lng": -3.761941,
                        },
                        "FR": Object {
                          "lat": 46.938698,
                          "lng": 2.399887,
                        },
                        "GB": Object {
                          "lat": 54.355481,
                          "lng": -2.278986,
                        },
                      }
        }
        markers={
                Array [
                        Object {
                          "position": Object {
                            "lat": 48.857767,
                            "lng": 2.293604,
                          },
                        },
                      ]
        }
        zoom={5} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Map {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "initialCenter": Object {
            "ES": Object {
              "lat": 40.481755,
              "lng": -3.761941,
            },
            "FR": Object {
              "lat": 46.938698,
              "lng": 2.399887,
            },
            "GB": Object {
              "lat": 54.355481,
              "lng": -2.278986,
            },
          },
          "markers": Array [
            Object {
              "position": Object {
                "lat": 48.857767,
                "lng": 2.293604,
              },
            },
          ],
          "zoom": 5,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <GoogleMap
          bootstrapURLKeys={
                    Object {
                              "key": "AIzaSyCYrFIvVAYqnWCNjjAfGIanmwxVvj1FPj8",
                            }
          }
          debounced={true}
          defaultCenter={
                    Object {
                              "lat": 54.355481,
                              "lng": -2.278986,
                            }
          }
          defaultZoom={5}
          distanceToMouse={[Function]}
          googleMapLoader={[Function]}
          hoverDistance={30}
          layerTypes={Array []}
          onChildClick={undefined}
          options={[Function]}
          style={
                    Object {
                              "height": "300px",
                              "position": "relative",
                              "width": "100%",
                            }
          }
          yesIWantToUseGoogleMapApiInternals={false}>
          <Mock
                    lat={48.857767}
                    lng={2.293604}
                    title={undefined}
                    trade={undefined} />
</GoogleMap>,
        "_debugID": 2,
        "_renderedOutput": <GoogleMap
          bootstrapURLKeys={
                    Object {
                              "key": "AIzaSyCYrFIvVAYqnWCNjjAfGIanmwxVvj1FPj8",
                            }
          }
          debounced={true}
          defaultCenter={
                    Object {
                              "lat": 54.355481,
                              "lng": -2.278986,
                            }
          }
          defaultZoom={5}
          distanceToMouse={[Function]}
          googleMapLoader={[Function]}
          hoverDistance={30}
          layerTypes={Array []}
          onChildClick={undefined}
          options={[Function]}
          style={
                    Object {
                              "height": "300px",
                              "position": "relative",
                              "width": "100%",
                            }
          }
          yesIWantToUseGoogleMapApiInternals={false}>
          <Mock
                    lat={48.857767}
                    lng={2.293604}
                    title={undefined}
                    trade={undefined} />
</GoogleMap>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Map
    initialCenter={
        Object {
            "ES": Object {
              "lat": 40.481755,
              "lng": -3.761941,
            },
            "FR": Object {
              "lat": 46.938698,
              "lng": 2.399887,
            },
            "GB": Object {
              "lat": 54.355481,
              "lng": -2.278986,
            },
          }
    }
    markers={
        Array [
            Object {
              "position": Object {
                "lat": 48.857767,
                "lng": 2.293604,
              },
            },
          ]
    }
    zoom={5} />,
}
`;
